server/test/repository/ExportServiceRepositoryTest.java

```diff
@@ -0,0 +1,137 @@
+    ImmutableList<QuestionOption> newOptionList =
+        ImmutableList.<QuestionOption>builder().addAll(currentOptions).add(newOption).build();
+    QuestionDefinition toUpdate =
+        new QuestionDefinitionBuilder(questionWeather.getQuestionDefinition())
+            .setQuestionOptions(newOptionList)
+            .build();
+    questionService.update(toUpdate);
+
+    ImmutableList<String> multiSelectHeaders =
+        repo.getMultiSelectedHeaders(questionWeather.getQuestionDefinition());
+    // not draft versions are not part of the header list
+    assertThat(multiSelectHeaders.size()).isEqualTo(3);
+    versionRepository.publishNewSynchronizedVersion();
+    ImmutableList<String> multiSelectHeaderUpdated =
+        repo.getMultiSelectedHeaders(questionWeather.getQuestionDefinition());
+    assertThat(multiSelectHeaderUpdated.size()).isEqualTo(4);
+    checkList(multiSelectHeaderUpdated, 0, "fall");
+    checkList(multiSelectHeaderUpdated, 1, "spring");
+    checkList(multiSelectHeaderUpdated, 2, "summer");
+    checkList(multiSelectHeaderUpdated, 3, "winter");
+  }
+
+  @Test
+  public void getMultiSelectedHeaders_ThrowsExceptionOnWrongQuestionType() {
+    QuestionDefinition questionDefinition =
+        testQuestionBank.applicantAddress().getQuestionDefinition();
+    assertThatThrownBy(() -> repo.getMultiSelectedHeaders(questionDefinition))
+        .isInstanceOf(RuntimeException.class)
+        .hasMessage("The Question Type is not checkbox");
+  }
+
+  @Test
+  public void getMultiSelectedHeaders_ThrowsExceptionOnUnpublishedQuestion() {
+    QuestionDefinitionConfig config =
+        QuestionDefinitionConfig.builder()
+            .setName("test")
+            .setDescription("test")
+            .setQuestionText(LocalizedStrings.of(Locale.US, "test"))
+            .setQuestionHelpText(LocalizedStrings.of(Locale.US, "This is sample help text."))
+            .build();
+    ImmutableList<QuestionOption> questionOptions =
+        ImmutableList.of(
+            QuestionOption.create(1L, 1L, "option1", LocalizedStrings.of(Locale.US, "option1")),
+            QuestionOption.create(2L, 2L, "option2", LocalizedStrings.of(Locale.US, "option2")),
+            QuestionOption.create(3L, 3L, "option3", LocalizedStrings.of(Locale.US, "option3")));
+    QuestionDefinition definition =
+        new MultiOptionQuestionDefinition(
+            config,
+            questionOptions,
+            MultiOptionQuestionDefinition.MultiOptionQuestionType.CHECKBOX);
+    assertThatThrownBy(() -> repo.getMultiSelectedHeaders(definition))
+        .isInstanceOf(NoSuchElementException.class)
+        .hasMessage("No value present");
+  }
+
+  private void checkList(ImmutableList<String> multiSelectHeaders, int index, String value) {
+    assertThat(multiSelectHeaders.contains(value)).isTrue();
+    assertThat(multiSelectHeaders.get(index)).isEqualTo(value);
+  }
+
+  private Question createMultiSelectQuestion(
```

Comment:
**praise**: Thanks for pulling this out into a helper function!

**suggestion (not blocking)**: structuring this using the Builder pattern would make this easier to extend or customize in the future (eg if we wanted more than 3 options):
https://github.com/civiform/civiform/blob/32bc2499819f2986eb15db46a31bfc7bc4a1cb18/server/test/services/export/AbstractExporterTest.java#L516-L569

Reply:
Can you file a bug and put it in the TODO, like `TODO(#5555)`?